### PR TITLE
foam-extend: add v5.0

### PR DESCRIPTION
**Pull request by kjrstory**
The current of foam-extend doesn't work properly. I've fixed it to make it work correctly.
Also added a new version in the process. 
I have tested versions 4.1 and 5.0 on gcc 9.4 cascadelake.

### CODE REVIEW

File: var/spack/repos/builtin/packages/foam-extend/package.py [hunk 1]
```diff
@@ -87,6 +87,7 @@ class FoamExtend(Package):
     # mgridgen is statically linked
     depends_on("parmgridgen", when="+parmgridgen", type="build")
     depends_on("paraview@:5.0.1", when="+paraview")
+    depends_on("mesquite")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/foam-extend/package.py:91]: Do you know if this dependency applies to other listed versions or for a specific version on?

File: var/spack/repos/builtin/packages/foam-extend/package.py [hunk 2]
```diff
@@ -53,12 +53,12 @@ class FoamExtend(Package):
...
     """
 
-    homepage = "http://www.extend-project.de/"
+    homepage = "https://sourceforge.net/projects/foam-extend/"
 
+    version("5.0", git="http://git.code.sf.net/p/foam-extend/foam-extend-5.0.git")
     version("4.1", git="http://git.code.sf.net/p/foam-extend/foam-extend-4.1.git")
     version("4.0", git="http://git.code.sf.net/p/foam-extend/foam-extend-4.0.git")
     version("3.2", git="http://git.code.sf.net/p/foam-extend/foam-extend-3.2.git")
-    version("3.1", git="http://git.code.sf.net/p/foam-extend/foam-extend-3.1.git")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/foam-extend/package.py:61]: We generally recommend deprecating versions before removing them.  Also, why remove 3.1 and not 3.0?  (Could use a comment to explain.)

Deprecation of old versions is discussed in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions.

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage (thanks for updating to `https`) and version git URL.  Do have some questions about deprecation and application of new changes.

File: var/spack/repos/builtin/packages/foam-extend/package.py (hunk 2 above)
Comment (kjrstory) [reply var/spack/repos/builtin/packages/foam-extend/package.py:61]: The first reason is that you cannot download the file from the address of 3.1. It is unclear whether the file was originally non-existent or has been deleted, but currently, there is no repository at the address of 3.1. 

To provide further clarification, there are two sourceforge projects related to *foam-extend*.
1. https://sourceforge.net/projects/openfoam-extend/
2. https://sourceforge.net/projects/foam-extend/

The first project includes versions 3.0 and 3.1, while the second project encompasses versions 3.0, 3.2, 4.0, 4.1, and 5.0.
I consider the first project to be deprecated, which is why I based my evaluation on the second project.

If I refer to the guide, in reality, even versions 3.0 and 3.2 are quite outdated and not commonly used. It appears that they would require patches to work with the current compiler versions, so considering them as deprecated seems accurate to me. What are your thoughts on this?

File: var/spack/repos/builtin/packages/foam-extend/package.py (hunk 1 above)
Comment (kjrstory) [reply var/spack/repos/builtin/packages/foam-extend/package.py:91]: In my opinion, it seems that *mesquite* has been needed from the early versions. I am conducting some testing on how the installation logs vary depending on the presence of *mesquite*, and once the testing is complete, I will let you know.

File: var/spack/repos/builtin/packages/foam-extend/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/foam-extend/package.py:61]: You can certainly mark them as deprecated then remove them after the next release.

**Comment by tldahlgren** [comment]
@kjrstory Ping me if the PR is ready to be merged as-is.

**Comment by tldahlgren** [comment]
@kjrstory Ping me when you're ready for this PR to merge.

**Comment by kjrstory** [comment]
@tldahlgren
Please review the merge again now.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**